app/javascript/packages/document-capture/components/in-person-location-step.tsx

```diff
@@ -181,35 +142,50 @@ function InPersonLocationStep({ onChange, toPreviousStep }) {
     [locationData, inProgress],
   );
 
+  const handleFoundAddress = useCallback((address) => {
+    setHasSearchedOnce(true);
+
+    setFoundAddress({
+      streetAddress: address.street_address,
+      city: address.city,
+      state: address.state,
+      zipCode: address.zip_code,
+    });
+  }, []);
+
   useEffect(() => {
     let mounted = true;
-    (async () => {
-      try {
-        const fetchedLocations = await getUspsLocations();
-
-        if (mounted) {
-          const formattedLocations = formatLocation(fetchedLocations);
-          setLocationData(formattedLocations);
-        }
-      } finally {
-        if (mounted) {
-          setIsLoadingComplete(true);
+    hasSearchedOnce &&
+      (async () => {
+        try {
+          const fetchedLocations = await getUspsLocations(foundAddress);
+
+          if (mounted) {
+            const formattedLocations = formatLocation(fetchedLocations);
+            setLocationData(formattedLocations);
+          }
+        } finally {
+          if (mounted) {
+            setIsLoadingComplete(true);
+          }
         }
-      }
-    })();
+      })();
     return () => {
       mounted = false;
```

Comment:
I think the variable could make sense as far as tracking whether the request is still valid by the time it finishes, as a way to avoid race conditions.

I've seen a similar suggestion from a React maintainer, called `didCancel` in their example: https://github.com/facebook/react/issues/14326#issuecomment-441680293